plugins/woocommerce/includes/wc-update-functions.php

```diff
@@ -2993,6 +2993,58 @@ function wc_update_961_migrate_default_email_base_color() {
 	}
 }
 
+/**
+ * Add old refunded order items to the product_lookup_table.
+ */
+function wc_update_1000_add_old_refunded_order_items_to_product_lookup_table() {
+	global $wpdb;
+
+	// Get every order ID where:
```

Comment:
Would it be worth testing this change in a large store to see if any performance issues arise? This could potentially affect query execution time and resource usage, as we will be retrieving all old refunded order items from the product lookup table.

Reply:
Yeah it did probably be worth testing, as I am looking at this logic, it also looks like this can probably be optimized quite a bit, as we are doing:
- Getting all the refunded order ids
- Then getting each order object individually ( triggering a query in behind )
- Checking if the order has line items ( this could be checked in the original query )
- Triggering a scheduled action for each refunded order ( this may not be as bad, as the scheduler will handle this ).

Reply:
So I did make a small update to this, so we only retrieve the entire order object if the `num_items_sold` is set to 0, this more accurately reflects a fully refunded order.
I don't anticipate the actual select query to be much of a bottleneck.

Plus this update function gets triggered in a cron job, same goes for the import calls, which in turn also gets badged if there is a bunch of them.